src/elements/content-sidebar/Sidebar.js

```diff
@@ -136,6 +112,18 @@ class Sidebar extends React.Component<Props, State> {
         history.push(`${history.location.pathname}/versions${fileVersionSlug}`);
     };
 
+    /**
+     * Getter for location state properties
+     *
+     * @param key - Optionally get a specific key value from state
+     * @returns {any} - The location state or state key value
+     */
+    getLocationState(key?: string): any {
+        const { location } = this.props;
+        const { state: locationState = {} } = location;
+        return key ? getProp(locationState, key) : locationState;
```

Comment:
This method seems a bit overloaded imo, does it return locationState properties or the location state itself? Maybe just keep the scope to properties and return null in this ternary?

Reply:
And do we need `getProp` here? Seems like the user defined properties you are adding to location state are only `silent` and `open`

Reply:
It accepts a single optional argument as a getter, so I'm not sure I agree it's too overloaded. I'll try to simplify it, though, since I'm not using the non-key aspect right now.